tests/platform_tests/test_auto_negotiation.py

```diff
@@ -186,239 +140,212 @@ def check_ports_up(duthost, dut_ports, expect_speed=None):
+        dut_advertised_speeds = 'all'
+    else:
+        dut_advertised_speeds = ','.join(duthost.get_supported_speeds(portname))
+
+    speed = enum_speed_per_dutport_fixture['speed']
+    skip_speed_if_unsupported(duthost, portname, speed)
+
     duthost.shell('config interface autoneg {} enabled'.format(dut_port))
     duthost.shell('config interface advertised-speeds {} {}'.format(dut_port, dut_advertised_speeds))
 
-    logger.info('Trying to get a common supported speed set among dut port, fanout port and cable')
-    supported_speeds = get_supported_speeds_for_port(duthost, dut_port, fanout, fanout_port)
-    pytest_require(supported_speeds, 'Ignore test for port {} due to cannot get supported speed for it'.format(dut_port))
-
-    logger.info('Run test based on supported speeds: {}'.format(supported_speeds))
     success = fanout.set_auto_negotiation_mode(fanout_port, True)
     pytest_require(success, 'Failed to set port autoneg on fanout port {}'.format(fanout_port))
-    
-    for speed in supported_speeds:
-        success = fanout.set_speed(fanout_port, speed)
-        pytest_require(success, 'Failed to advertised speeds on fanout port {}, speed {}'.format(fanout_port, speed))
-
-        logger.info('Wait until the port status is up, expected speed: {}'.format(speed))
-        wait_result = wait_until(SINGLE_PORT_WAIT_TIME, 
-                                PORT_STATUS_CHECK_INTERVAL, 
-                                0, 
-                                check_ports_up, 
-                                duthost, 
-                                [dut_port], 
-                                speed)
-
-        pytest_assert(wait_result, '{} are still down. Advertised speeds: DUT = {}, fanout = {}'
-            .format(dut_port, dut_advertised_speeds, speed))
-        fanout_actual_speed = fanout.get_speed(fanout_port)
-        pytest_assert(fanout_actual_speed == speed, 'expected fanout speed: {}, but got {}'.format(speed, fanout_actual_speed))
-
-def test_force_speed(enum_dut_portname_module_fixture):
+    success = fanout.set_speed(fanout_port, speed)
+    pytest_require(success, 'Failed to advertised speeds on fanout port {}, speed {}'.format(fanout_port, speed))
+
+    logger.info('Wait until the port status is up, expected speed: {}'.format(speed))
+    wait_result = wait_until(
+        SINGLE_PORT_WAIT_TIME,
+        PORT_STATUS_CHECK_INTERVAL,
+        0,
+        check_ports_up,
+        duthost,
+        [dut_port],
+        speed)
+
+    pytest_assert(
+        wait_result, '{} are still down. Advertised speeds: DUT = {}, fanout = {}'
+        .format(dut_port, dut_advertised_speeds, speed))
+    fanout_actual_speed = fanout.get_speed(fanout_port)
+    pytest_assert(fanout_actual_speed == speed, 'expected fanout speed: {}, but got {}'.format(speed, fanout_actual_speed))
+
+
+FEC_FOR_SPEED = {
+    25000: 'fc',
+    50000: 'fc',
+    100000: 'rs',
+    200000: 'rs',
+    400000: 'rs'
```

Comment:
why do we need these hard coded FEC for different speed?

Reply:
peers might try to auto-set different FECs for a given speed which will cause force_speed test to fail. So I decided to set both speed and FEC explicitly